src/components/setup-sharehouse/SetupContents.tsx

```diff
@@ -0,0 +1,54 @@
+import { Stepper } from '@/components/ui/stepper';
+import { Button } from '@/components/ui/button';
+import { ReactNode } from 'react';
+
+interface ISetupContentsProps {
+  title: string;
+  currentStep: number;
+  maxSteps: number;
+  onNext: () => void;
+  onBack?: () => void;
+  children?: ReactNode;
+  isNextDisabled?: boolean;
+  onOpen?: () => void;
+}
+
+const SetupContents = ({
+  title,
+  currentStep,
+  maxSteps,
+  onNext,
+  onBack,
+  children,
+  isNextDisabled,
+  onOpen,
+}: ISetupContentsProps) => {
+  return (
+    <div className="flex-1 flex flex-col justify-between h-full">
+      <div>
+        <h1 className="my-4">{title}</h1>
+        <Stepper currentStep={currentStep} maxSteps={maxSteps} />
+
+        {children}
+      </div>
+      <div className="flex justify-center gap-4 py-4">
+        {onBack && (
+          <Button variant={'tertiary'} onClick={onBack}>
+            Back
+          </Button>
+        )}
+        {onOpen ? (
+          <Button onClick={onOpen} disabled={isNextDisabled}>
+            Next
+          </Button>
+        ) : (
+          <Button onClick={onNext} disabled={isNextDisabled}>
+            Next
+          </Button>
```

Comment:
Sure, here's the translation:

---

It seems that the button disable functionality is not working. Could you please check it?

There is a prop called "isNextDisabled," but it looks like it's not being passed correctly from the parent component. Additionally, when I check `isNextDisabled` with `console.log`, it shows 'undefined'.

<img width="146" alt="Screenshot 2024-07-25 at 5 20 39 PM" src="https://github.com/user-attachments/assets/41b93c1d-e163-44b2-8bda-89a88277bee1">

Reply:
I'd like to add `disable` functionality after implementing POST. So I'd like to keep this props as it is. What do you think?